### DISCUSSION TITLE
On-demand ISR fails with custom domain on Vercel

### DISCUSSION THREAD

**Discussion by amirardalan** [vercel/community#560] [Help] [answered]
I have a Next.js app hosted on Vercel. I'm using the new Next.js On-demand ISR feature. ISR works great on localhost and also on my vercel app domain (https://amirardalan.vercel.app) but the same `unstable_revalidate` fetch request fails from my custom domain name (https://amirardalan.com). The domain is connected through Vercel and it is assigned to my app on production.

I have triple checked that my environment variable for the secret is configured correctly on all instances including production, and I redeployed with no build cache.

Am I missing something simple here? Any help is greatly appreciated!

**Comment by leerob** [comment]
If you add logging to `getStaticProps`, do you see that being ran when you hit the revalidate route?

**Reply by amirardalan** [reply]
@leerob I see the log on the page in question _after_ successfully hitting the revalidate API endpoint on amirardalan.vercel.app but not on amirardalan.com using the same test case.

The revalidate API call itself fails when hitting from custom domain. Behaves as expected on my vercel subdomain...

**Comment by amirardalan** [comment, marked as answer]
I was able to track down and resolve this issue; I am using Next Preview mode in conjunction with on-demand ISR, this is where the problem occurred.

**SOLUTION:** I swapped the order of my functions to Exit Next Preview Mode _before_ fetching the `revalidate` webhook. Error is now gone. I am guessing this has something to do with existing cache invalidation from preview mode conflicting with ISR.

The discrepancy between domains was simply because I had an authenticated session with preview mode active only on my primary domain. Mystery solved.

Such an awesome feature, thanks for all the hard work on it!
